lib/pages/getStaticPropsWithRevalidate/redirects.js

```diff
@@ -1,26 +1,53 @@
 const { join } = require("path");
```

Comment:
The easiest/most clear approach IMO is to actually not mess with `redirects.js`, but to simply modify `pages.js` and add this at the end:

```js
  // Add route without locale for supporting naked routes in i18n
  if (isRouteWithDefaultLocale(route)) {
    pages.push({
      route: getRouteWithoutLocale(route),
      srcRoute: srcRoute,
      dataRoute: getDataRouteWithoutLocale(route),
    });
  }
```

`setup.js` already makes sure we only set up one function for every srcRoute and `redirects.js` will set up the correct redirects for all routes and data routes pointing to the `srcRoute`.

Reply:
thought about this for awhile and really have no problem moving the logic over. what's holding me back is that i feel like it belongs more in redirects because it is exactly that - a redirect. there's no page for the locale-less route, but there is (and should/will be, regardless), a redirect. i think it's misplacing that logic.

that said, if we were doing the `copy` strategy (instead of redirects), _then_ it would make sense to me to modify pages.